### PR TITLE
DCS-906: two utility functions for hashing

### PR DESCRIPTION
**Pull request by GurnankCheema**

### CODE REVIEW

File: server/utils/utils.test.js [hunk 1]
```diff
@@ -95,3 +103,16 @@ describe('parseDate', () => {
...
+describe('hash', () => {
+  it('should return hash', () => {
+    // result generated via https://approsto.com/sha-generator/
+    const result = stringToHash('abcABC')
+    expect(result).toEqual('XVnsPWdyGrzz4FzZhZQ1s2agjqxtfKqd3tNx1JC1tM8=')
+  })
+
+  it('should verify hash', () => {
+    const result = isHashedString('XVnsPWdyGrzz4FzZhZQ1s2agjqxtfKqd3tNx1JC1tM8=', 'abcABC')
+    expect(result).toBe(true)
```
Comment (andrewrlee) [inline server/utils/utils.test.js:116]: probably also want to test negative cases

File: server/utils/utils.js [hunk 2]
```diff
@@ -42,6 +43,14 @@ const properCaseFullName = name =>
         .map(properCaseName)
         .join(' ')
 
+const stringToHash = inputString => {
+  return crypto.createHash('sha256').update(inputString).digest('base64')
+}
+
+const isHashedString = (hashValue, originalString) => {
+  return hashValue === crypto.createHash('sha256').update(originalString).digest('base64')
```
Comment (andrewrlee) [inline server/utils/utils.js:51]: might be better to use stringToHash? 

missing type signatures too on these.

File: server/utils/utils.js [hunk 3]
```diff
@@ -42,6 +43,14 @@ const properCaseFullName = name =>
         .map(properCaseName)
         .join(' ')
 
+const stringToHash = inputString => {
+  return crypto.createHash('sha256').update(inputString).digest('base64')
```
Comment (andrewrlee) [inline server/utils/utils.js:47]: maybe force passing a salt here?

File: server/utils/utils.test.js (hunk 1 above)
Comment (GurnankCheema) [reply server/utils/utils.test.js:116]: yes

**Review by louisenorris — DISMISSED** [review]
LGTM

File: server/utils/utils.js (hunk 3 above)
Comment (andrewrlee) [reply server/utils/utils.js:47]: or read the salt value from config here so the calling code doesn't need to pass it?

File: server/utils/utils.js (hunk 2 above)
Comment (GurnankCheema) [reply server/utils/utils.js:51]: changed module from js to ts
above changes made

File: server/utils/utils.js (hunk 3 above)
Comment (GurnankCheema) [reply server/utils/utils.js:47]: ok. Have passed salt in a separate. Have also changed method name to isHashOfString as is more descriptive

File: server/utils/utils.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+export const stringToHash = (inputString: string, salt: string): string => {
+  return crypto.createHash('sha256').update(`${inputString}${salt}`).digest('base64')
+}
+
+export const isHashOfString = (hashValue: string, originalString: string, salt: string): boolean => {
+  return hashValue === stringToHash(originalString, salt)
+}
+
+module.exports = {
```
Comment (andrewrlee) [inline server/utils/utils.ts:59]: think this can be deleted?  not used with typescript import/export

File: server/utils/utils.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
+import R from 'ramda'
+import crypto from 'crypto'
+import moment, { Moment } from 'moment'
+
+export const isNilOrEmpty = R.either(R.isEmpty, R.isNil)
+export const getIn = R.path
+export const { equals } = R
+export const firstItem = R.head
+export const mergeWithRight = R.mergeDeepRight
```
Comment (andrewrlee) [inline server/utils/utils.ts:9]: think we can get rid of `mergeWithRight` as not being used

Comment (andrewrlee) [reply server/utils/utils.ts:9]: and `getIn`

File: server/utils/utils.ts (hunk 4 above)
Comment (GurnankCheema) [reply server/utils/utils.ts:59]: aha!

File: server/utils/utils.ts (hunk 5 above)
Comment (GurnankCheema) [reply server/utils/utils.ts:9]: good idea. I didn't check for their use across that app

**Review by andrewrlee — APPROVED** [review]
LGTM
